pkg/kubelet/collector/interfaces.go

```diff
@@ -0,0 +1,67 @@
+	// Starts the collector process if necessary, e.g. when using the builtin cadvisor.
+	// Otherwise, does nothing and just returns.
+	Start() error
+
+	// Returns the basic node spec, e.g. can be used to report to k8s master the
+	// basic information about a node.
+	NodeInfo() (*NodeInfo, error)
+
+	// Returns the software version of various components of a node, e.g. can be
+	// used to report back to k8s master the versions of relevant software on a node.
+	VersionInfo() (*VersionInfo, error)
+
+	// Returns the file system information of a given label or mount point (mount point
+	// is represented in absolute path, i.e. starting with '/', and label is not), e.g.
+	// can be used by kubelet to check if there's enough disk space to schedule new pods
+	// and by the image manager to decide if garbage collection should start.
+	FsInfo(fsLabel string) (*FsInfo, error)
+
+	// Returns a channel for watching requested event types, e.g. can be
+	// used by oom watcher to detect oom events.
+	WatchEvents(request *EventRequest) (chan *Event, error)
+
+	// Get raw metrics of a Docker container
+	// containerID: Docker container ID
+	// numStats: max number of metrics to return
+	// start: start time to query the metrics data, and if omitted, the beginning of time is assumed
+	// end: end time to query the metrics data, and if omitted, the current time is assumed
+	// Returned value is a map of container name to its metrics data, and this data can be in a
+	// collector-specific format, e.g. can be used to handle /stats/... REST calls
+	DockerContainerInfo(containerID string, numStats int, start time.Time, end time.Time) (map[string]interface{}, error)
+
+	// Get raw metrics of a Linux container
+	// containerName: absolute path of a Linux container
+	// numStats: max number of metrics to return
+	// start: start time to query the metrics data, and if omitted, the beginning of time is assumed
+	// end: end time to query the metrics data, and if omitted, the current time is assumed
+	// subcontainer: if true, returns raw metrics information of this container and its child containers
+	RawContainerInfo(containerName string, numStats int, start time.Time, end time.Time, subcontainers bool) (map[string]interface{}, error)
+
+	// DockerContainerInfo() and RawContainerInfo() return collector-specific raw data, which need to be
+	// parsed to retrieve a particular metric type, e.g. cpu, memory, cpu load, file system. The respective
+	// parsing interface functions, e.g. GetContainerCPUInfo(), GetContainerMemoryInfo(), GetContainerNetworkInfo(),
+	// etc., should be defined to retrieve such data in a format consumeable by Kubernetes. However, as of now
+	// Kubernetes is currently not using any of such metrics data, thus, these interface functions (along with the
+	// corresponding data structs) should be defined and tuned at the time when such code is added.
+
+	// TODO: Define parsing interface functions as use cases are more concrete
```

Comment:
maybe it would be good to define an empty interface type as a placehold for this? e.g. 
```
type ContainerInfo interface{}
```